src/vivarium_public_health/metrics/mortality.py

```diff
@@ -7,121 +7,173 @@
 excess mortality in the simulation.
 
 """
+from collections import Counter
+from typing import Dict, List
+
+import pandas as pd
+
+from vivarium.framework.engine import Builder, ConfigTree
+from vivarium.framework.event import Event
+from vivarium.framework.population import PopulationView
+from vivarium.framework.time import Timedelta
+
 from vivarium_public_health.disease import DiseaseState, RiskAttributableDisease
-from vivarium_public_health.metrics.utilities import (
-    get_age_bins,
-    get_deaths,
-    get_person_time,
-    get_years_of_life_lost,
-)
+from vivarium_public_health.metrics.stratification import ResultsStratifier
+from vivarium_public_health.utilities import to_time_delta
 
 
 class MortalityObserver:
     """An observer for cause-specific deaths, ylls, and total person time.
 
-    By default, this counts cause-specific deaths, years of life lost, and
-    total person time over the full course of the simulation. It can be
-    configured to bin these measures into age groups, sexes, and years
-    by setting the ``by_age``, ``by_sex``, and ``by_year`` flags, respectively.
+    By default, this counts cause-specific deaths and years of life lost over
+    the full course of the simulation. It can be configured to add or remove
+    stratification groups to the default groups defined by a ResultsStratifier.
 
     In the model specification, your configuration for this component should
     be specified as, e.g.:
 
     .. code-block:: yaml
 
         configuration:
-            metrics:
+            observers:
                 mortality:
-                    by_age: True
-                    by_year: False
-                    by_sex: True
+                    exclude:
+                        - "year"
+                    include:
+                        - "death_year"
```

Comment:
I think you'll want to write a document about the default + include/exclude logic used to determine what is stratified by what and then link to it in these observer docstrings.  Otherwise this code example is not possible to interpret cause there's no context for how it's used.

Reply:
Also, `"death_year"` should have gone away to be replaced by `"year"` and `"cause_of_death"` is no longer a stratification level.

Reply:
Good call on documentation. death_year I fixed in a later PR, but I'll move it into this one.